### PR TITLE
IFU Cherrypick into rocm6.1_internal_testing over Dec4 upstream commit

### PR DESCRIPTION
**Pull request by pruthvistony**

### CODE REVIEW

**Comment by pruthvistony** [comment]
Build triggered - http://rocm-ci.amd.com/job/mainline-framework-pytorch-ci/1503/

File: aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLibBlas.cpp [hunk 1]
```diff
@@ -225,7 +225,15 @@ void triangular_solve_batched_cublas(const Tensor& A, const Tensor& B, bool left
 
 template <typename scalar_t>
 inline void apply_gels_batched(const Tensor& A, Tensor& B, Tensor& infos) {
+#if not defined(ROCM_VERSION)
```
Comment (jeffdaily) [inline aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLibBlas.cpp:228]: is that valid syntax?

Comment (pruthvistony) [reply aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLibBlas.cpp:228]: Nice catch.
Not sure how this is handled by the pre processor. I will correct it.

File: torch/csrc/jit/codegen/fuser/codegen.cpp [hunk 2]
```diff
@@ -677,6 +677,23 @@ std::string generateKernel(
   // Still need the key defined, but empty.
   env.s("RuntimeHeader", R"()");
 #endif
+
+#if ROCM_VERSION >= 60000
```
[the remark this answers is not in the document]
Comment (pruthvistony) [reply torch/csrc/jit/codegen/fuser/codegen.cpp:681]: I believe this change is backward compatible, the change is building fine on ROCm mainline and ROCm 6.0 release builds. So I think this change is already part of ROCm and I believe it is OK to keep it >= 60000.

**Comment by pruthvistony** [comment]
Build successful - http://rocm-ci.amd.com/job/mainline-framework-pytorch-ci/1505/

**Comment by pruthvistony** [comment]
https://amdcloud-my.sharepoint.com/:x:/r/personal/pmagundu_amd_com/_layouts/15/doc2.aspx?sourcedoc=%7B2c1b2bbc-6f31-4cc3-8e97-be74d273821d%7D&action=edit&activeCell=%27CP_internal_testing_branch%27!E64&wdinitialsession=d09fc073-a2ae-42e8-b4e3-30674e8bc192&wdrldsc=6&wdrldc=1&wdrldr=AccessTokenExpiredWarning%2CRefreshingExpiredAccessT

**Comment by pruthvistony** [comment]
> @pruthvistony Looks good to me if the commits are all cherry-picks from rocm6.0_internal_testing branch. Only feedback is:
> 
>     1. We should start using `cherry-pick -x` to have a trace of which commit it was cherry-picked from
> 
>     2. Are all of these cherry-picks added to the spreadsheet you were maintaining to track upstreaming of various fixes in branch?

--- Didnt know about `cherry-pick -x` , we should follow it. I will update the confluence page.
--- Yes, the excel sheet is updated with the cherry-pick done and which commits are upstream/ignored/skipped.

**Review by jithunnair-amd — APPROVED** [review]

**Pull request merged**